src/FreeformIdeas.js

```diff
@@ -1,90 +1,173 @@
-import React, { useCallback, useState } from "react";
+import React, { useCallback, useReducer, useState } from "react";
 import { useDrop } from "react-dnd";
 import { Idea } from "./Idea";
 import { ItemTypes } from "./ItemTypes";
 import update from "immutability-helper";
 import { v4 as uuidv4 } from "uuid";
 import Modal from "./Modal";
 import IdeaForm from "./IdeaForm";
+import Arrow from "./Arrow";
 
 const FreeFormIdeas = () => {
-  const [ideas, setIdeas] = useState({
-    a: {
-      id: "a",
-      top: 20,
-      left: 80,
-      title: "Here is an Example to get you Started",
+  const [ideas, ideasDispatch] = useReducer(
+    (ideas, action) => {
+      switch (action.type) {
+        case "update":
+          return update(ideas, {
+            [action.id]: {
+              $merge: action.data,
+            },
+          });
+        case "create":
+          return update(ideas, {
+            [action.id]: {
+              $set: Object.assign({ id: action.id, linkList: [] }, action.data),
+            },
+          });
+        default:
+          throw new Error(`Unexpected action type: ${action.type}`);
+      }
     },
-    b: { id: "b", top: 180, left: 20, title: "Great Idea!" },
-  });
+    {
+      a: {
+        id: "a",
+        top: 20,
+        left: 80,
+        title: "Here is an Example to get you Started",
+        linkList: ["b"],
+      },
+      b: { id: "b", top: 180, left: 20, title: "Great Idea!", linkList: [] },
+    }
+  );
+
+  const [coords, setCoords] = useState({ top: 0, left: 0 });
 
   const [showIdeaModal, setShowIdeaModal] = useState(false);
   const ideaModalShow = useCallback(() => setShowIdeaModal(true), []);
   const ideaModalHide = useCallback(() => setShowIdeaModal(false), []);
 
-  const [selectedIdea, setSelectedIdea] = useState(ideas.a);
-  const setSelected = useCallback((id) => setSelectedIdea(ideas[id]), [ideas]);
+  const [selectedId, setSelectedId] = useState("a");
+  const selectedIdea = ideas[selectedId];
+
+  const [linkerEnd, setLinkerEnd] = useState();
+  const [linkingState, setLinkingState] = useState(false);
+  const linkBreak = useCallback(() => setLinkingState(false), []);
+
+  const onLinkStart = useCallback((e) => {
+    setLinkerEnd([e.clientY - 65, e.clientX]);
+    setLinkingState(true);
+  }, []);
+
+  const adjustLinker = useCallback(
+    (e) => {
+      if (linkingState) {
+        setLinkerEnd([e.clientY - 65, e.clientX]);
+      }
```

Comment:
```suggestion
      if (!linkingState) return
      setLinkerEnd([e.clientY - 65, e.clientX]);
```

Reply:
Is there more to this than just for the sake of getting rid of a line?

Reply:
Nope. It's just good to get into the habit using early returns to reduce indentation.